File: src/battleships-api/Models/MatchSettings.cs
namespace Models;

public enum GameMode {
    Classic,
    Ammo,
    FogOfWar,
}

public class MatchSettings
{
    public GameMode GameMode { get; set; }
    public bool UseDice { get; set; }

    public MatchSettings() {
        this.GameMode = GameMode.Classic;
        this.UseDice = false;
    }
}

File: src/battleships-ui/src/models/MatchSettings.ts
export enum GameMode {
  Classic = 'Classic',
  Ammo = 'Ammo',
  FogOfWar = 'FogOfWar',
}

export default class MatchSettings {
  gameMode!: GameMode;
  useDice!: boolean;

  constructor(object?: Partial<MatchSettings>) {
    this.gameMode = object?.gameMode ?? GameMode.Classic;
    this.useDice = !!object?.useDice;
  }
}
